meilisearch-http/src/routes/setting.rs

```diff
@@ -1,107 +1,569 @@
-use std::collections::HashMap;
-
-use http::StatusCode;
-use serde::{Deserialize, Serialize, Deserializer};
-use tide::response::IntoResponse;
-use tide::{Context, Response};
+use meilisearch_core::settings::{Settings, SettingsUpdate, UpdateState};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet, HashSet};
+use tide::{Request, Response};
 
 use crate::error::{ResponseError, SResult};
-use crate::helpers::tide::ContextExt;
+use crate::helpers::tide::RequestExt;
 use crate::models::token::ACL::*;
 use crate::routes::document::IndexUpdateResponse;
 use crate::Data;
 
-#[derive(Default, Serialize, Deserialize)]
+pub async fn get_all(ctx: Request<Data>) -> SResult<Response> {
+    ctx.is_allowed(SettingsRead)?;
+    let index = ctx.index()?;
+    let db = &ctx.state().db;
+    let reader = db.main_read_txn()?;
+
+    let stop_words_fst = index.main.stop_words_fst(&reader)?;
+    let stop_words = stop_words_fst.unwrap_or_default().stream().into_strs()?;
+    let stop_words: BTreeSet<String> = stop_words.into_iter().collect();
+    let stop_words = if stop_words.len() > 0 {
+        Some(stop_words)
+    } else {
+        None
+    };
+
+    let synonyms_fst = index.main.synonyms_fst(&reader)?.unwrap_or_default();
+    let synonyms_list = synonyms_fst.stream().into_strs()?;
+
+    let mut synonyms = BTreeMap::new();
+
+    let index_synonyms = &index.synonyms;
+
+    for synonym in synonyms_list {
+        let alternative_list = index_synonyms.synonyms(&reader, synonym.as_bytes())?;
+
+        if let Some(list) = alternative_list {
+            let list = list.stream().into_strs()?;
+            synonyms.insert(synonym, list);
+        }
+    }
+
+    let synonyms = if synonyms.len() > 0 {
+        Some(synonyms)
+    } else {
+        None
+    };
+
+    let ranking_rules = match index.main.ranking_rules(&reader)? {
+        Some(rules) => Some(rules.iter().map(|r| r.to_string()).collect()),
+        None => None,
+    };
+    let ranking_distinct = index.main.ranking_distinct(&reader)?;
+
+    let schema = index.main.schema(&reader)?;
+
+    let attribute_identifier = schema.clone().map(|s| s.identifier());
+    let attributes_searchable = schema.clone().map(|s| s.get_indexed_name());
+    let attributes_displayed = schema.clone().map(|s| s.get_displayed_name());
```

Comment:
Is it really necessary to clone the schema each time?

Reply:
Yes, apparently I should. schema is moved each time